File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.cs
using System;
using System.Text;

public class Solution {
    public string solution(string code) 
    {
        var sb = new StringBuilder();

        bool mode = false;
        for(int i = 0; i < code.Length; ++i)
        {
            if(code[i] == '1')
                mode = !mode;
            else if((mode && i % 2 == 1) || (!mode && i % 2 == 0))
                sb.Append(code[i]);
        }
        
        return sb.ToString() == string.Empty ? "EMPTY" : sb.ToString();
    }
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string code) {
    string answer = "";
    
    int mode = 0;
    for (int i = 0; i < code.length(); ++i)
    {
        char c = code[i];
        
        if (c == '1')
        {
            mode = 1 - mode;
            continue;
        }
        
        if (mode == 0 && i % 2 == 0 
            || mode == 1 && i % 2 == 1)
        {
            answer += c;
        }
    }
    
    return answer.length() != 0 ? answer : "EMPTY";
}